File: playground/linkedList.cpp
#include <iostream>

template <typename T>
struct Node
{
    T data;
    Node *next;
};

template <typename T>
class LinkedList
{
private:
    int listLength = 0;
    Node<T> *head = NULL;

public:
    // LinkedList()
    // {
    // }
    void append(T data)
    {
        listLength += 1;
        Node<T> *addingNode = (Node<T> *)malloc(sizeof(Node<T>));
        addingNode->data = data;
        addingNode->next = head;
        head = addingNode;
    }
    int pop()
    {
        int returnData = head->data;
        Node<T> *nextHead = head->next;
        free(head);
        head = nextHead;
        return returnData;
    }
    void printList()
    {
        Node<T> *start = head;
        for (int i = 0; i < listLength; i++)
        {
            // printf("%d->", start->data);
            std::cout << start->data << "->";
            start = start->next;
        }
        printf("NULL");
        printf("\n");
    }
};

int main(int argc, char const *argv[])
{
    LinkedList<std::string> test;
    test.append("Hello");
    test.append("World");
    test.append("!");
    test.append("!");
    test.printList();
    return 0;
}

File: playground/linkedList.py
from typing import Any


class Node:
    def __init__(self, data: Any):
        self.next = None
        self.data = data


class linkedList:
    def __init__(self):
        self.headNode = Node(None)
        self.__len__ = 0

    def append(self, data: Any):
        newNode = Node(data)
        newNode.next = self.headNode
        self.headNode = newNode
        self.__len__ += 1

    def printList(self):
        currentNode = self.headNode
        for i in range(self.__len__):
            print(currentNode.data)
            currentNode = currentNode.next

    def pop(self):
        returnData = self.headNode.data
        self.headNode = self.headNode.next
        self.__len__ -= 1
        return returnData

    def size(self):
        return self.__len__


testlist = linkedList()
testlist.append(1)
testlist.append(2)
testlist.append(3)
testlist.append(4)
testlist.pop()
testlist.printList()
